### PR TITLE
Add whitening

### LINKED ISSUE #14: Whitening Transform Subset of Population
Add option to whiten using only control samples. (only a subset of samples and apply to remaining samples)

### PR DESCRIPTION
**Pull request by gwaygenomics**
closes #14

### CODE REVIEW

**Comment by gwaygenomics** [comment]
cc @hkhawar - whitening now implemented!

File: pycytominer/cyto_utils/transform.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    def fit(self, X, y=None):
+        """
+        Identify the whitening transform given self.X
+
+        Argument:
+        X - dataframe to fit whitening transform
+        """
+        # Get the mean of the features (columns) and center if specified
+        self.mu = X.mean()
```
Comment (shntnu) [inline pycytominer/cyto_utils/transform.py:35]: @gwaygenomics did you mean to write `self.mu = np.mean(X, 0)` ?

Comment (gwaygenomics) [reply pycytominer/cyto_utils/transform.py:35]: is it the same thing?

Comment (shntnu) [reply pycytominer/cyto_utils/transform.py:35]: `self.mu = X.mean(0)` returns a vector: the mean of `X` along `axis=0`
`self.mu = X.mean()` returns a scalar: the mean of flattened `X`.

Comment (shntnu) [reply pycytominer/cyto_utils/transform.py:35]: PS – please don't ruin your weekend pondering this. I added this note so I don't forget. Too bad GitHub doesn't have scheduled send :D

**Comment by gwaygenomics** [comment]
Is X in your example a pandas.DataFrame?

File: pycytominer/cyto_utils/transform.py (hunk 1 above)
Comment (shntnu) [reply pycytominer/cyto_utils/transform.py:35]: > Is X in your example a pandas.DataFrame?

Ah no – it's a numpy array. You're good.

**Pull request merged**